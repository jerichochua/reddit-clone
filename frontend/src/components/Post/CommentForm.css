.comment-form-container {
  margin-top: -1px;
  border: 1px solid rgb(235, 237, 240);
  transition: border 0.1s ease-in-out;
}

.comment-form-container:hover {
  border: 1px solid rgb(0, 149, 246);
}

.comment-form-container:focus-within {
  border: 1px solid rgb(0, 149, 246);
}

.comment-form {
  display: flex;
  flex-direction: column;
}

.comment-form-textarea {
  width: 100%;
  padding: 8px;
  background-color: rgb(250, 250, 250);
  font-size: 15px;
  resize: none;
  outline: none;
  border: none;
  border-bottom: 1px solid rgb(235, 237, 240);
}

.comment-form-button {
  padding: 8px 24px;
  margin: 4px;
  font-size: 12px;
  font-weight: 700;
  text-transform: uppercase;
  border: none;
  border-radius: 4px;
  background-color: rgb(0, 149, 246);
  color: white;
  cursor: pointer;
  align-self: flex-end;
}
