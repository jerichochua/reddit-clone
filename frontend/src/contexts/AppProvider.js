import React, { createContext, useContext, useReducer } from 'react';
import jwt_decode from 'jwt-decode';

const AppContext = createContext();
AppContext.displayName = 'AppContext';

const token = localStorage.getItem('token');
const userId = token ? jwt_decode(token).id : null;
const username = token ? jwt_decode(token).username : null;

const initialState = {
  userId,
  username,
  token,
};

const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_USER':
      return {
        ...state,
        userId: action.payload.userId,
        username: action.payload.username,
      };
    case 'SET_TOKEN':
      return { ...state, token: action.payload };
    case 'LOGOUT':
      return { ...state, userId: null, username: null, token: null };
    default:
      return state;
  }
};

export const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export const useAppContext = () => {
  return useContext(AppContext);
};
