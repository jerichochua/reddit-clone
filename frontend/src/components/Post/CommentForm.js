import React, { useState } from 'react';
import { post } from '../../util/api';
import { FormButton } from '../Form';
import { useAppContext } from '../../contexts/AppProvider';
import './CommentForm.css';

const CommentForm = ({ postId }) => {
  const { state } = useAppContext();
  const [message, setMessage] = useState('');

  const onSubmitForm = async (e) => {
    e.preventDefault();
    const body = { content: e.target.content.value };
    try {
      const response = await post(`posts/${postId}`, body, state.token);
      if (response) {
        setMessage('comment added');
        e.target.content.value = '';
      } else {
        setMessage('error adding comment');
        console.error(response);
      }
    } catch (error) {
      setMessage(`error adding comment: ${error.message}`);
    }
  };

  return (
    <div className='comment-form-container'>
      <form className='comment-form' onSubmit={onSubmitForm}>
        <textarea
          name='content'
          className='comment-form-textarea'
          placeholder='Add a comment...'
          rows='2'
          required
        ></textarea>
        <div className='comment-form-footer'>
          {message && <span className='comment-footer-message'>{message}</span>}
          <FormButton label='Comment' type='submit' />
        </div>
      </form>
    </div>
  );
};

export default CommentForm;
