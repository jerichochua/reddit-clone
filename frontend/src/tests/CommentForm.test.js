import renderer from 'react-test-renderer';
import { AppProvider } from '../contexts/AppProvider';
import CommentForm from '../components/Post/CommentForm';

it('renders correctly', () => {
  const component = renderer.create(
    <AppProvider>
      <CommentForm postId={1} />
    </AppProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
