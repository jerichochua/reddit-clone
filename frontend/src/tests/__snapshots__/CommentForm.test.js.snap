// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="comment-form-container"
>
  <form
    className="comment-form"
    onSubmit={[Function]}
  >
    <textarea
      className="comment-form-textarea"
      name="content"
      placeholder="Add a comment..."
      required={true}
      rows="2"
    />
    <div
      className="comment-form-footer"
    >
      <button
        className="form-btn"
        type="submit"
      >
        Comment
      </button>
    </div>
  </form>
</div>
`;
